### PR TITLE
IALERT-3360 saml sign assertions

### PR DESCRIPTION
**Pull request by mau-synopsys**
-Perform manual signature verification as the functionality was turned off by default in 5.7.x
-Remove usage of wantAuthnRequests as that is different from wantAssertionsSigned
-On send auth event, email address is user name if it contains '@'
-Convert AlertRelyingPartyRegistrationRepository to component

### CODE REVIEW

File: authentication-saml/src/main/java/com/synopsys/integration/alert/authentication/saml/security/SAMLSignedAssertionsValidator.java [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+    public ValidationResult validateSignedAssertions(OpenSaml4AuthenticationProvider.AssertionToken assertionToken) {
+        Optional<SAMLConfigModel> optionalSAMLConfigModel = configAccessor.getConfiguration();
+
+        if (optionalSAMLConfigModel.isPresent()) {
+            boolean wantAssertionsSigned = BooleanUtils.toBoolean(optionalSAMLConfigModel.get().getWantAssertionsSigned());
+            String alertRegisteredAssertingPartyId = alertRelyingPartyRegistrationRepository.findByRegistrationId("default").getAssertingPartyDetails().getEntityId();
+
+            if (wantAssertionsSigned
+                && !(assertionToken.getAssertion().isSigned()
+                     && alertRegisteredAssertingPartyId.equals(assertionToken.getAssertion().getIssuer().getValue()))
```
Comment (ChomickiM) [inline authentication-saml/src/main/java/com/synopsys/integration/alert/authentication/saml/security/SAMLSignedAssertionsValidator.java:31]: I feel like this should be collapsed into a single line, or if its too long we could split out the components and save them as booleans and compare that way

Comment (mau-synopsys) [reply authentication-saml/src/main/java/com/synopsys/integration/alert/authentication/saml/security/SAMLSignedAssertionsValidator.java]: Updated, the alertRegisteredAssertingPartyId check is no longer necessary.

**Review by ChomickiM — APPROVED** [review]

File: authentication-saml/src/main/java/com/synopsys/integration/alert/authentication/saml/security/SAMLManager.java [hunk 2]
```diff
@@ -10,10 +10,14 @@
...
 import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
 import org.springframework.security.converter.RsaKeyConverters;
 import org.springframework.security.saml2.core.Saml2X509Credential;
 import org.springframework.security.saml2.provider.service.registration.RelyingPartyRegistration;
+import org.springframework.security.saml2.provider.service.registration.RelyingPartyRegistrationRepository;
 import org.springframework.security.saml2.provider.service.registration.RelyingPartyRegistrations;
+import org.springframework.security.saml2.provider.service.web.DefaultRelyingPartyRegistrationResolver;
+import org.springframework.security.saml2.provider.service.web.RelyingPartyRegistrationResolver;
 import org.springframework.security.saml2.provider.service.web.authentication.OpenSaml4AuthenticationRequestResolver;
```
Comment (DanaMaxfield) [inline authentication-saml/src/main/java/com/synopsys/integration/alert/authentication/saml/security/SAMLManager.java:21]: Why the added imports if code using these wasn't also added?

Comment (mau-synopsys) [reply authentication-saml/src/main/java/com/synopsys/integration/alert/authentication/saml/security/SAMLManager.java]: Updated, test artifacts

**Review by psantos1113 — APPROVED** [review]

**Pull request merged**